### PR TITLE
Add swagger support and allow cors

### PR DESCRIPTION
**Pull request by baoj2010**

### CODE REVIEW

File: putt/settings.py [hunk 1]
```diff
@@ -39,9 +39,13 @@
     'django.contrib.staticfiles',
     'rest_framework',
     'rest_framework_simplejwt',
+    'drf_spectacular',
+    'corsheaders',
+    'projects',
```
Comment (JoshYuJump) [inline putt/settings.py]: `projects ` should be included like `users.apps.UsersConfig`,  and the order had to change to from above.

File: putt/settings.py [hunk 2]
```diff
@@ -130,4 +134,44 @@
     'DEFAULT_AUTHENTICATION_CLASSES': (
         'rest_framework_simplejwt.authentication.JWTAuthentication',
     ),
+    "DEFAULT_PAGINATION_CLASS": "rest_framework.pagination.PageNumberPagination",
+    "PAGE_SIZE": 10,
+    "DEFAULT_SCHEMA_CLASS": "drf_spectacular.openapi.AutoSchema",
+
 }
+# 跨域允许的请求方式，可以使用默认值，默认的请求方式为:
```
Comment (JoshYuJump) [inline putt/settings.py]: Comments should using English

File: requirements.txt [hunk 3]
```diff
@@ -2,3 +2,6 @@ Django>=3.2,<4.0
 djangorestframework>=3.13.1,<4.0
 djangorestframework-simplejwt
 django-filter>=21.1,<22.0
+drf-spectacular>=0.21.1,<1.0
```
Comment (JoshYuJump) [inline requirements.txt]: `drf-spectacular>=0.21.1,<1.0` should move down after `django-cors-headers>=3.10.1,<4.0`

File: templates/swagger-ui.html [hunk 4]
```diff
@@ -0,0 +1,31 @@
+<!DOCTYPE html>
```
Comment (JoshYuJump) [inline templates/swagger-ui.html:1]: `templates/swagger-ui.html` should remove as `drf_spectacular` included

File: putt/settings.py (hunk 1 above)
Comment (baoj2010) [reply putt/settings.py]: resolved

File: putt/settings.py (hunk 2 above)
Comment (baoj2010) [reply putt/settings.py]: resolved

File: requirements.txt (hunk 3 above)
Comment (baoj2010) [reply requirements.txt]: resolved

File: templates/swagger-ui.html (hunk 4 above)
Comment (baoj2010) [reply templates/swagger-ui.html:1]: It can not be dropped

**Pull request merged**